qiskit/providers/ibmq/experiment/experiment.py

```diff
@@ -45,7 +46,9 @@ def __init__(
             analysis_results: Optional[List[AnalysisResult]] = None,
             hub: Optional[str] = None,
             group: Optional[str] = None,
-            project: Optional[str] = None
+            project: Optional[str] = None,
+            visibility: Union[ExperimentVisibility, str] = None,
```

Comment:
I'd prefer naming this `share_level` to match `IBMQJob`. And this should probably default to `private` to make it more obvious to the user what the default is.

Reply:
>I'd prefer naming this share_level to match IBMQJob.

Yeah I can do that.

>And this should probably default to private to make it more obvious to the user what the default is.

I started with that based on how the analysis results quality field has a default, but then I thought about how the update function works and it looks like that would always pass the visibility value (or quality in the analysis result case) on update even if we don't want to change the value. I'm not sure if that could really happen, i.e. would someone construct an Experiment object directly with the fields set to update (uuid to identify the object and let's say end_time to update) and then if visibility defaults to `private` in the object but I didn't want to update the visibility to `private` (let's say the experiment visibility is `public`), they'd get the implicit update without thinking about it. If that's not really a scenario that should happen then I can set a default, IOW, would anyone only ever update an Experiment after first retrieving it and updating the fields from remote data?

Reply:
The scenario you're describing is that a person would manually construct an `Experiment` object with an existing experiment uuid and default the visibility to private. They might then inadvertently overwrite the existing visibility value when saving the experiment in results db. I don't think it's a likely scenario but still possible. I suppose you can default it to `None` and describe in the docstring that visibility defaults to private for new experiments, but if they're creating an `Experiment` to represent an existing one (they really shouldn't), then the visibility field is None as it's unknown.

A better solution would probably be removing `experiment_uuid` from `Experiment` constructor so users cannot manually construct an existing experiment (i.e. they can only retrieve it). But that's more outside the scope of this PR.

Reply:
> I suppose you can default it to `None` and describe in the docstring that visibility defaults to private for new experiments, but if they're creating an `Experiment` to represent an existing one (they really shouldn't), then the visibility field is None as it's unknown.

This was the course I was thinking of taking.

Reply:
One downside of defaulting to `None` is that if I create a new `Experiment` and do `experiment.share_level`, it'll give me `None`, and I'd just have to know that means private. There are ways around it but probably not worth the effort as not allowing constructing an `Experiment` with a uuid is still the better option.